pkg/app/api/grpcapi/api.go

```diff
@@ -482,9 +508,16 @@ func (a *API) GetPlanPreviewResults(ctx context.Context, req *apiservice.GetPlan
 			)
 			return nil, status.Error(codes.Internal, fmt.Sprintf("Failed to decode output data of command %s", commandID))
 		}
+
 		results = append(results, &result)
 	}
 
+	// All URL fields inside the result model are empty.
+	// So we fill them before sending to the client.
+	for _, r := range results {
+		r.FillURLs(a.webBaseURL)
```

Comment:
nit: I thought putting this inside right after the loop is enough (like you did before commit 😄 ). But if you intend to be more clear, it's on you

Reply:
I will keep this as it is because there are 2 loops (L480, L512) that are updating this list.

Reply:
That is true 👍